docbrown/db/src/graph.rs

```diff
@@ -116,8 +119,8 @@ impl Graph {
         self.shards.iter().map(|shard| shard.out_edges_len()).sum()
     }
 
-    pub fn has_edge(&self, src: u64, dst: u64) -> bool {
-        self.shards.iter().any(|shard| shard.has_edge(src, dst))
+    pub fn has_edge(&self, v1: u64, v2: u64) -> bool {
```

Comment:
I think we need to benchmark this as it seems like we are going to be doing a linear search every time we look up, making algos which use it O(n^2)